backend/app/analysis/find_vanishing_point.py

```diff
@@ -0,0 +1,220 @@
+"""
+
+find_vanishing_point.py
+
+analysis to find significant lines in image and filter lines to find place where most lines
+intersect, which should be the vanishing point
+"""
+
+import numpy as np
+import cv2
+
+# from matplotlib import pyplot as plt
+from app.models import Photo
+
+MODEL = Photo
+
+
+def analyze(photo: Photo):
+    """
+    Given a photo, returns the coordinate of the vanishing point,
+    where the vanishing point is the point that has the minimum sum of
+    distances to all detected lines in the photo
+    """
+    image = photo.get_image_data()
+    # Convert image to grayscale
+    # (Changes image array shape from (height, width, 3) to (height, width))
+    # (Pixels (image[h][w]) will be a value from 0 to 255)
+    grayscale_image = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+    lines = auto_canny(grayscale_image)
+
+    # plt.subplot(121), plt.imshow(lines[0], cmap='gray')
+    # plt.title('Edge Image'), plt.xticks([]), plt.yticks([])
+
+    filter_lines = []
+    # Filters out vertical and horizontal lines
+    for line in lines[1]:
+        try:
+            point_1_x, point_1_y, point_2_x, point_2_y = line[0]
+
+            # find angle of line from horizontal
+            theta = abs(np.arctan((point_2_y - point_1_y) / (point_2_x - point_1_x)))
+
+            # only put line in filter_lines if the angle is NOT within epsilon of
+            # horizontal/vertical
+            epsilon = np.pi / 16
+            if (point_2_x - point_1_x) != 0 and abs(theta - np.pi / 2) > 2 * epsilon and theta > \
+                    epsilon:
+                # cv2.line(image, (point_1_x, point_1_y), (point_2_x, point_2_y), (0, 0, 255), 3, 8)
+                filter_lines.append(line)
+        except ZeroDivisionError:
+            pass
+    # van_point = find_van_coord(filter_lines, image.shape[0], image.shape[1])
```

Comment:
Here's an example of what I'm talking about in the review message -- all of this is commented out and really should be deleted, as it makes the routine difficult to read.